internal/otlp/json_decoder.go

```diff
@@ -34,24 +34,18 @@ func newJSONDecoder() *jsonDecoder {
 
 func (d *jsonDecoder) DecodeLogs(buf []byte) (interface{}, error) {
 	ld := &otlpcollectorlogs.ExportLogsServiceRequest{}
-	if err := d.delegate.Unmarshal(bytes.NewReader(buf), ld); err != nil {
-		return nil, err
-	}
-	return ld, nil
+	err := d.delegate.Unmarshal(bytes.NewReader(buf), ld)
+	return ld, err
```

Comment:
This is not totally equivalent to the old code but I guess it is fine to return non-nil with an error. Should this behavior be documented in the DecodeLogs interface definition?